README.md

```diff
@@ -987,10 +993,80 @@ have a single leading `-` or this will result in failures. For example,
 ### Bash Completion
 
 You can enable completion commands by setting the `EnableBashCompletion`
-flag on the `App` object.  By default, this setting will only auto-complete to
-show an app's subcommands, but you can write your own completion methods for
-the App or its subcommands.
+flag on the `App` object to `true`.  By default, this setting will allow 
+auto-completion for an app's subcommands, but you can write your own completion 
+methods for the App or its subcommands as well.
 
+#### Default auto-completion
+
+```go
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/urfave/cli"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.EnableBashCompletion = true
+	app.Commands = []cli.Command{
+		{
+			Name:    "add",
+			Aliases: []string{"a"},
+			Usage:   "add a task to the list",
+			Action: func(c *cli.Context) error {
+				fmt.Println("added task: ", c.Args().First())
+				return nil
+			},
+		},
+		{
+			Name:    "complete",
+			Aliases: []string{"c"},
+			Usage:   "complete a task on the list",
+			Action: func(c *cli.Context) error {
+				fmt.Println("completed task: ", c.Args().First())
+				return nil
+			},
+		},
+		{
+			Name:    "template",
+			Aliases: []string{"t"},
+			Usage:   "options for task templates",
+			Subcommands: []cli.Command{
+				{
+					Name:  "add",
+					Usage: "add a new template",
+					Action: func(c *cli.Context) error {
+						fmt.Println("new task template: ", c.Args().First())
+						return nil
+					},
+				},
+				{
+					Name:  "remove",
+					Usage: "remove an existing template",
+					Action: func(c *cli.Context) error {
+						fmt.Println("removed task template: ", c.Args().First())
+						return nil
+					},
+				},
+			},
+		},
+	}
+
+	err := app.Run(os.Args)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+```
+![](https://media.giphy.com/media/S3gyjumR56IpDyH23r/giphy.gif)
```

Comment:
⭐⭐⭐ _(must change)_ gif links to an external location make me uncomfortable - upload these to our github repo please!

Reply:
Ya I totally agree, any preference on where to store them? I can make an "example_gifs" directory in the "autocomplete" directory?  It doesn't look like there's an obvious existing directory to drop them in.

Reply:
Put them in this path:

/docs/images/*.gif